src/components/vm/nics/nicBody.tsx

```diff
@@ -266,3 +300,611 @@ export const NetworkTypeAndSourceRow = ({
         </>
     );
 };
+
+export function portForwardText(pf: VMInterfacePortForward): string {
+    let text = "";
+
+    if (pf.address)
+        text += pf.address + " ";
+    if (pf.dev)
+        text += pf.dev + " ";
+
+    if (!pf.ranges.some(r => r.exclude != "yes"))
+        text += _("all ports");
+    for (let i = 0; i < pf.ranges.length; i++) {
+        const r = pf.ranges[i];
+        if (text != "")
+            text += ", ";
+        if (r.exclude == "yes")
+            text += _("excluding ");
+        text += r.start;
+        if (r.end)
+            text += "-" + r.end;
+        if (r.to)
+            text += " → " + r.to + (r.end ? "-" + (Number(r.to) + Number(r.end) - Number(r.start)).toString() : "");
+        if (pf.proto && pf.proto != "tcp")
+            text += "/" + pf.proto;
```

Comment:
I really dislike having one-letter variables and would much rather want to keep it expanded a bit